packages/builder/src/settings/_components/CloneResourcesModal.svelte

```diff
@@ -0,0 +1,282 @@
+      [ResourceType.QUERY]: [],
+      [ResourceType.AUTOMATION]: [],
+      [ResourceType.WORKSPACE_APP]: [],
+      [ResourceType.SCREEN]: [],
+    }
+    modal.show()
+  }
+
+  interface DataType {
+    _id: string
+    name: string
+    direct: boolean
+    __disabled: boolean
+  }
+
+  let toWorkspaceId: string | undefined
+
+  let selectedResources: Record<ResourceType, DataType[]> = {
+    [ResourceType.DATASOURCE]: [],
+    [ResourceType.TABLE]: [],
+    [ResourceType.ROW_ACTION]: [],
+    [ResourceType.QUERY]: [],
+    [ResourceType.AUTOMATION]: [],
+    [ResourceType.WORKSPACE_APP]: [],
+    [ResourceType.SCREEN]: [],
+  }
+
+  $: workspaces = $appsStore.apps
+    .filter(a => a.devId !== sdk.applications.getDevAppID($appStore.appId))
+    .sort((a, b) => a.name.localeCompare(b.name))
+
+  function onShow() {}
+
+  async function onConfirm() {
+    await API.resource
+      .duplicateResourceToWorkspace({
+        resources: resourcesToBeCopied.map(r => r._id),
+        toWorkspace: toWorkspaceId!,
+      })
+      .then(() => {
+        notifications.success("Resources copied successfully")
+        isOpen = false
+        modal.hide()
+      })
+      .catch(err => {
+        notifications.error(err.message)
+      })
+  }
+
+  let dependantResources: Record<
+    string,
+    Partial<Record<ResourceType, UsedResource[]>>
+  > = {}
+
+  const mapToDataType = (d: AnyDocument): DataType => ({
+    _id: d._id!,
+    name: d.name,
+    direct: true,
+    __disabled: false,
+  })
+
+  $: $tables.list.forEach(t => rowActions.refreshRowActions(t._id!))
+
+  let resourceTypesToDisplay: {
+    [K in ResourceType]: {
+      displayName: string
+      type: K
+      data: DataType[]
+    }
+  }
+  $: resourceTypesToDisplay = {
+    [ResourceType.WORKSPACE_APP]: {
+      displayName: "Apps",
+      data: $workspaceAppStore.workspaceApps.map(a =>
+        mapToDataType({ ...a, name: `${a.name} (${a.screens.length} screens)` })
+      ),
+      type: ResourceType.WORKSPACE_APP,
+    },
+    [ResourceType.TABLE]: {
+      displayName: "BB tables",
+      data: $tables.list.map(mapToDataType),
+      type: ResourceType.TABLE,
+    },
+    [ResourceType.DATASOURCE]: {
+      displayName: "Datasources",
+      data: $datasources.list.map(mapToDataType),
+      type: ResourceType.DATASOURCE,
+    },
+    [ResourceType.AUTOMATION]: {
+      displayName: "Automations",
+      data: $automationStore.automations.map(mapToDataType),
+      type: ResourceType.AUTOMATION,
+    },
+    [ResourceType.QUERY]: {
+      displayName: "Queries",
+      data: $queries.list.map(mapToDataType),
+      type: ResourceType.QUERY,
+    },
+    [ResourceType.ROW_ACTION]: {
+      displayName: "Row actions",
+      data: [],
+      type: ResourceType.ROW_ACTION,
+    },
+    [ResourceType.SCREEN]: {
+      displayName: "Screens",
+      data: [],
+      type: ResourceType.SCREEN,
+    },
+  }
+
+  $: isOpen &&
+    API.resource
+      .searchForUsage()
+      .then(res => {
```

Comment:
In general, why do your prefer `.then` rather than async / wait and wrapping in a try / catch? 

I think that's more prevalent in general across our codebase, it's NAB. I'm just curious.

Reply:
I usually prefer async/await, but in these cases, we can't use awaits at the root. We could always add an async function that handles this, but I think it's cleaner to read this way